File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/cart/cartBookResponse.ts
import { BookResponse, CartBook, mapBookResponseToBook } from "../../../..";

export interface CartBookResponse {
    id: string;
    bookAmount: number;
    bookId: number;
    book: BookResponse;
}

export function mapCartBookResponseToCartBook(response: CartBookResponse): CartBook {
    return {
        id: response?.id,
        bookAmount: response?.bookAmount,
        bookId: response?.bookId,
        book: mapBookResponseToBook(response?.book)
    }
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Shop/CartBookResponse.cs
﻿using LibraryShopEntities.Domain.Dtos.Library;

namespace LibraryShopEntities.Domain.Dtos.Shop
{
    public class CartBookResponse
    {
        public string? Id { get; set; }
        public int BookAmount { get; set; }
        public int BookId { get; set; }
        public BookResponse? Book { get; set; }
    }
}
